src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1389,79 +1389,127 @@ def get_network_lb(cli_ctx, resource_group_name, lb_name):
 
 
 def process_vmss_create_namespace(cmd, namespace):
+    from azure.cli.core.azclierror import InvalidArgumentValueError
     # uniform_str = 'Uniform'
     flexible_str = 'Flexible'
     if namespace.orchestration_mode.lower() == flexible_str.lower():
-        validate_tags(namespace)
-        if not namespace.location:
-            get_default_location_from_resource_group(cmd, namespace)
+
         # The commentted parameters are also forbidden, but they have default values.
         # I don't know whether they are provided by user.
+
+        namespace.load_balancer_sku = 'Standard'  # lb sku MUST be standard
+        # namespace.public_ip_per_vm = True  # default to true for VMSS Flex
+        # namespace.disable_overprovision = True  # overprovisioning must be false for vmss flex preview
+        # namespace.single_placement_group = False  # SPG must be false for VMSS flex
+        namespace.upgrade_policy_mode = None
+        namespace.use_unmanaged_disk = None
+
         banned_params = [
-            namespace.accelerated_networking,
-            namespace.admin_password,
+            # namespace.accelerated_networking,
+            # namespace.admin_password,
             # namespace.admin_username,
-            namespace.application_gateway,
+            # namespace.application_gateway,
             # namespace.app_gateway_capacity,
             # namespace.app_gateway_sku,
-            namespace.app_gateway_subnet_address_prefix,
-            namespace.application_security_groups,
-            namespace.assign_identity,
-            namespace.authentication_type,
-            namespace.backend_pool_name,
-            namespace.backend_port,
-            namespace.computer_name_prefix,
-            namespace.custom_data,
-            namespace.data_caching,
-            namespace.data_disk_sizes_gb,
+            # namespace.app_gateway_subnet_address_prefix,
+            # namespace.application_security_groups,
+            # namespace.assign_identity,
+            # namespace.authentication_type,
+            # namespace.backend_pool_name,
+            # namespace.backend_port,
+            # namespace.computer_name_prefix,
+            # namespace.custom_data,
```

Comment:
May I ask whether parameter `user_data` is also supported in Flexible orchestration mode?

Reply:
user_data is not supported in this version of `vmss create`. You could check when you add the support.

Reply:
Got it, thanks~